packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -0,0 +1,744 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+import {
+	AnchorNode,
+	EmptyKey,
+	FieldKey,
+	MapTree,
+	TreeNodeSchemaIdentifier,
+	TreeValue,
+	Value,
+} from "../../core/index.js";
+import { brand, fail, mapIterable } from "../../util/index.js";
+import {
+	FlexTreeContext,
+	FlexTreeEntityKind,
+	FlexTreeField,
+	FlexTreeFieldNode,
+	FlexTreeLeafNode,
+	FlexTreeMapNode,
+	FlexTreeNode,
+	FlexTreeNodeEvents,
+	FlexTreeOptionalField,
+	FlexTreeRequiredField,
+	FlexTreeSequenceField,
+	FlexTreeTypedField,
+	FlexTreeTypedNode,
+	FlexTreeTypedNodeUnion,
+	FlexTreeUnboxField,
+	FlexTreeUnboxNodeUnion,
+	FlexibleFieldContent,
+	FlexibleNodeSubSequence,
+	TreeStatus,
+	flexTreeMarker,
+	indexForAt,
+} from "../flex-tree/index.js";
+import {
+	FlexAllowedTypes,
+	FlexFieldNodeSchema,
+	FlexFieldSchema,
+	FlexMapNodeSchema,
+	FlexTreeNodeSchema,
+	LeafNodeSchema,
+	isLazy,
+	schemaIsFieldNode,
+	schemaIsLeaf,
+	schemaIsMap,
+	schemaIsObjectNode,
+} from "../typed-schema/index.js";
+import { FlexImplicitAllowedTypes, normalizeAllowedTypes } from "../schemaBuilderBase.js";
+import { FlexFieldKind } from "../modular-schema/index.js";
+import { FieldKinds, SequenceFieldEditBuilder } from "../default-schema/index.js";
+import { createEmitter } from "../../events/index.js";
+
+// #region Nodes
+
+/** A node's parent field and its index in that field */
+interface LocationInField {
+	readonly parent: MapTreeField<FlexAllowedTypes>;
+	readonly index: number;
+}
+
+/**
+ * A readonly implementation of {@link FlexTreeNode} which wraps a {@link MapTree}.
+ * @remarks Any methods that would mutate the node will fail,
+ * as will the querying of data specific to the {@link LazyTreeNode} implementation (e.g. {@link FlexTreeNode.context}).
+ * MapTreeNodes are unconditionally cached -
+ * when retrieved via {@link getOrCreateMapTreeNode}, the same {@link MapTree} object will always produce the same `MapTreeNode` object.
+ */
+export class MapTreeNode<TSchema extends FlexTreeNodeSchema> implements FlexTreeNode {
+	public readonly [flexTreeMarker] = FlexTreeEntityKind.Node as const;
+
+	private readonly events = createEmitter<FlexTreeNodeEvents>();
+
+	/**
```

Comment:
Nit: overall, the tooling better supports the pattern of appending tsdoc comments to each of the inline properties, rather than documenting the constructor with `@param` comments for them.

Intellisense should still display param-level docs for the properties if they are documented this way, and the generated documentation story works out better as well.

Reply:
Good call - I moved the doc comments around.